TanukiKit/Projects.swift

```diff
@@ -0,0 +1,221 @@
+        accessLevel = json["access_level"] as? Int
+        notificationLevel = json["notification_level"] as? Int
+    }
+}
+
+public extension TanukiKit {
+    /**
+     Fetches the Projects for the current user
+     - parameter page: Current page for project pagination. `1` by default.
+     - parameter perPage: Number of projects per page. `100` by default.
+     - parameter archived: Limit by archived status. Default is `""`, set to `true` to enable.
+     - parameter visibility: Limit by visibility `public`, `internal` or `private`. Default is `""`
+     - parameter orderBy: Return projects ordered by `id`, `name`, `path`, `created_at`, `updated_at`, or `last_activity_at` fields. Default is `created_at`.
+     - parameter sort: Return projects sorted in asc or desc order. Default is `desc`.
+     - parameter search: Return list of authorized projects matching the search criteria. Default is `""`
+     - parameter simple: Return only the ID, URL, name, and path of each project. Default is `""`, set to `true` to enable.
+     - parameter completion: Callback for the outcome of the fetch.
+     */
+    public func projects(_ session: RequestKitURLSession = URLSession.shared, page: String = "1", perPage: String = "20", archived: String = "", visibility: String = "", orderBy: String = "created_at", sort: String = "desc", search: String = "", simple: String = "", completion: @escaping (_ response: Response<[Project]>) -> Void) -> URLSessionDataTaskProtocol? {
+        let router = ProjectRouter.readAuthenticatedProjects(configuration, page, perPage, archived, visibility, orderBy, sort, search, simple)
+        return router.loadJSON(session, expectedResultType: [[String: AnyObject]].self) { json, error in
+            if let error = error {
+                completion(Response.failure(error))
+            }
+
+            if let json = json {
+                let projects = json.map { Project($0) }
+                completion(Response.success(projects))
+            }
+        }
+    }
+}
+
+// MARK: Router
+
+enum ProjectRouter: Router {
+    case readAuthenticatedProjects(Configuration, String, String, String, String, String, String, String, String)
+
+    var configuration: Configuration {
+        switch self {
+        case .readAuthenticatedProjects(let config, _, _, _, _, _, _, _, _): return config
+        }
+    }
+
+    var method: HTTPMethod {
+        return .GET
+    }
+
+    var encoding: HTTPEncoding {
+        return .url
+    }
+
+    var params: [String: Any] {
+        switch self {
+        case .readAuthenticatedProjects(_, let page, let perPage, let archived, let visibility, let orderBy, let sort, let search, let simple):
+            return ["per_page": perPage as AnyObject, "page": page as AnyObject, "archived": archived as AnyObject, "visibility": visibility as AnyObject, "order_by": orderBy as AnyObject, "sort": sort as AnyObject, "search": search as AnyObject, "simple": simple as AnyObject]
```

Comment:
No, but the `params` expects `[String: Any]` you are explicitly converting to `[String: AnyObject]` which is ok but could be omitted. If you feel like it you can change it throughout the projects __in a separate PR__.